modules/@apostrophecms/doc-type/index.js

```diff
@@ -147,9 +147,7 @@ module.exports = {
         }
       },
       afterArchive: {
-        // Mark draft only after moving to trash, to reactivate UI
-        // associated with things never published before
-        async markNeverPublishedAndRevertDraftToPublished(req, doc) {
+        async retainOnlyAsDraft(req, doc) {
```

Comment:
This covers what it's really for, everything it does is part of realizing a situation in which it appears as a new draft.